drivers/soundwire/intel.c

```diff
@@ -1005,9 +1005,13 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct sn
 {
 	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
 	struct sdw_intel *sdw = cdns_to_intel(cdns);
+	struct sdw_intel_link_res *res = sdw->link_res;
 	struct sdw_cdns_dma_data *dma;
 	int ret = 0;
 
+	if (res->ops && res->ops->trigger)
+		res->ops->trigger(dai, cmd, substream->stream);
+
```

Comment:
you really need to explain what would happen in the case of TRIGGER_SUSPEND. Are there going to be **TWO** callbacks doing something, or is this only for regular start/stop triggers?

Reply:
and you really need to explain why we need this for IPC4, this is far from obvious: we didn't need this before.

Reply:
Comments added.